src/collective/behavior/amp/utils.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+from collective.behavior.amp.config import AMP_VALID_ATTRIBUTES
+from lxml import html
+
+
+class Html2Amp:
+    def fix_img_tags(self, el):
+        """Change <img> tags into <amp-img> tags with layout responsive.
+        :param el: [required] LXML element to be sanitized.
+        :type el: lxml.html.HtmlElement
+        """
+        for tag in el.xpath('//img'):
+            tag.tag = 'amp-img'
+            tag.attrib['layout'] = 'responsive'
+
+    def remove_invalid_tags(self, el):
+        """Remove invalid tags.
+        :param el: [required] LXML element to be sanitized.
+        :type el: lxml.html.HtmlElement
+        """
+        for tag in el.iterdescendants():
+            if tag.tag in AMP_VALID_ATTRIBUTES:
+                continue
+            parent = tag.getparent()
+            parent.remove(tag)
+
+    def __call__(self, code):
+        el = html.fromstring(code)
+
+        # lxml automatically add a <span> around tags without parent
```

Comment:
The result of RichText field is a list of ``<p>`` tags without any tag around.

When LXML process these tags, it add a ``<span>`` tag around automatically.

I'm changing this ``<span>`` tag into ``<div class="amp-text">``.